packages/flutter/lib/src/material/date_picker_theme.dart

```diff
@@ -163,14 +163,59 @@ class DatePickerThemeData with Diagnosticable {
   /// grid of the date picker.
   ///
   /// This will be used instead of the color provided in [dayStyle].
+  ///
+  /// This supports different colors based on the [WidgetState]s of
+  /// the day button, such as [WidgetState.selected], [WidgetState.hovered],
+  /// [WidgetState.focused], and [WidgetState.disabled].
+  ///
+  /// ```dart
+  /// dayBackgroundColor: WidgetStateProperty.resolveWith((Set<WidgetState> states) {
+  ///   if (states.contains(WidgetState.selected)) {
+  ///     return Theme.of(context).colorScheme.primary;
+  ///   }
+  ///   return Colors.transparent;
+  /// })
+  /// ```
```

Comment:
Should probably include a comment that mentions dayOverlayColor. Maybe something like this at the bottom:

```dart 
See also:
  * [dayOverlayColor] which applies an overlay over the day labels depending on the [WidgetState].
```
Or a comment up above.